lib/entities/content-type.js

```diff
@@ -223,7 +228,39 @@ function createContentTypeApi (http) {
      * @func isDraft
      * @return {boolean}
      */
-    isDraft: createDraftChecker()
+    isDraft: createDraftChecker(),
+
+    /**
+     * Omits and deletes a field if it exists on the contentType. This is a convenience method which does both operations at once and potentially less
+     * safe than the standard way. See note about deleting fields on the Update method.
+     * @memberof ContentType
+     * @func omitAndDeleteField
+     * @return {Promise<ContentType>} Object returned from the server with updated metadata.
+     */
+    omitAndDeleteField: function (id) {
+      return this.findAndUpdateField(id, 'omitted', true)
+        .then(newContentType => newContentType.findAndUpdateField(id, 'deleted', true))
+        .catch(errorHandler)
+    },
+
+    /**
+     * @private
+     * @param {string} id - unique ID of the field
+     * @param {string} key - the attribute on the field to change
+     * @param {string} value - the value to set the attribute to
+     * @return {Promise<ContentType>}
+     */
+    findAndUpdateField: function (id, key, value) {
+      const index = _.findIndex(this.fields, {'id': id})
+      if (index >= 0) {
+        const newField = this.fields[index]
+        newField[key] = value
+        this.fields.splice(index, 1, newField)
+        return this.update()
+      } else {
```

Comment:
You won't need this else here due to the return in the line above. The parser will never reach the second return anyways.

This is called "return early" (http://blog.timoxley.com/post/47041269194/avoid-else-return-early)